src/openrct2/ride/Vehicle.cpp

```diff
@@ -7650,54 +7642,53 @@ void Vehicle::UpdateReverserCarBogies()
  * @param z (dx)
  * @param otherVehicleIndex (bp)
  */
-static bool vehicle_update_motion_collision_detection(
-    Vehicle* vehicle, int16_t x, int16_t y, int16_t z, uint16_t* otherVehicleIndex)
+bool Vehicle::UpdateMotionCollisionDetection(const CoordsXYZ& loc, uint16_t* otherVehicleIndex)
 {
-    if (vehicle->UpdateFlag(VEHICLE_UPDATE_FLAG_1))
+    if (UpdateFlag(VEHICLE_UPDATE_FLAG_1))
         return false;
 
-    auto vehicleEntry = vehicle->Entry();
+    auto vehicleEntry = Entry();
     if (vehicleEntry == nullptr)
     {
         return false;
     }
 
     if (!(vehicleEntry->flags & VEHICLE_ENTRY_FLAG_BOAT_HIRE_COLLISION_DETECTION))
     {
-        vehicle->var_C4 = 0;
+        var_C4 = 0;
 
         // If hacking boat hire rides you can end up here
         if (otherVehicleIndex == nullptr)
             return false;
 
         Vehicle* collideVehicle = GET_VEHICLE(*otherVehicleIndex);
 
-        if (vehicle == collideVehicle)
+        if (this == collideVehicle)
             return false;
 
-        int32_t x_diff = abs(x - collideVehicle->x);
+        int32_t x_diff = abs(loc.x - collideVehicle->x);
         if (x_diff > 0x7FFF)
             return false;
 
-        int32_t y_diff = abs(y - collideVehicle->y);
+        int32_t y_diff = abs(loc.y - collideVehicle->y);
         if (y_diff > 0x7FFF)
             return false;
 
-        int32_t z_diff = abs(z - collideVehicle->z);
+        int32_t z_diff = abs(loc.z - collideVehicle->z);
         if (x_diff + y_diff + z_diff > 0xFFFF)
             return false;
 
-        uint16_t ecx = std::min(vehicle->var_44 + collideVehicle->var_44, 560);
+        uint16_t ecx = std::min(var_44 + collideVehicle->var_44, 560);
         ecx = ((ecx >> 1) * 30) >> 8;
 
         if (x_diff + y_diff + z_diff >= ecx)
             return false;
 
-        uint8_t direction = (vehicle->sprite_direction - collideVehicle->sprite_direction + 7) & 0x1F;
+        uint8_t direction = (sprite_direction - collideVehicle->sprite_direction + 7) & 0x1F;
         return direction < 0xF;
     }
 
-    auto location = CoordsXY{ x, y };
+    auto location = CoordsXY{ loc.x, loc.y };
```

Comment:
```suggestion
    auto location = loc;
```

Reply:
location is CoordsXY, while loc is CoordsXYZ. Won't that be a problem?

Reply:
Hmm, then you can do:
`auto location = CoordsXY{loc}`